### PR TITLE
(feat)Line: add support for hiding line charts by default

### PR DESCRIPTION
**Pull request by clcatalan**
Solves issue for: https://github.com/plouc/nivo/issues/2481

Solution is for user to simply pass an array of keys in the initialHiddenIds props for the Line Chart. This  toggles the legend to "off" by default. The app consuming nivo should do the computation to determine if a key should be hidden or not. 

*this is what it looks like on load, the cognac line is hidden by default
Screenshot 2024-03-13 at 11 38 01 PM

### CODE REVIEW

**Comment by jerichocalibuso** [comment]
Wow galing

**Comment by plouc** [comment]
Thank you for this @clcatalan! Could you please add tests for this new feature?

**Comment by clcatalan** [comment]
hi @plouc added unit tests, all Line unit tests still pass as expected
Screenshot 2024-06-17 at 1 28 14 PM

Screenshot 2024-06-17 at 1 28 40 PM

File: packages/line/tests/Line.test.js [hunk 1]
```diff
@@ -89,6 +90,44 @@ it('should create slice for each x value', () => {
...
+        },
+    ]
+    const wrapper = mount(
+        <Line 
+            width={500} 
+            height={300} 
+            data={data}  
+            enableSlices="x" 
+            animate={false} 
+            initialHiddenIds={['B']}
```
Comment (plouc) [inline packages/line/tests/Line.test.js:123]: I think we should also test that we can hide several series.

**Review by plouc — CHANGES_REQUESTED** [review]

File: packages/line/src/Line.js [hunk 2]
```diff
@@ -107,6 +107,7 @@ const Line = props => {
         enableTouchCrosshair = false,
 
         role = 'img',
+        initialHiddenIds = [],
```
Comment (plouc) [inline packages/line/src/Line.js:110]: Sorry, but I missed 2 things during my initial review:
- It needs to be added to the types as this package hasn't been migrated to TypeScript yet
- This property should also be added to the documentation (https://github.com/plouc/nivo/blob/master/website/src/data/components/line/props.ts)

**Comment by clcatalan** [comment]
unit test for hiding multiple series by default
Screenshot 2024-07-11 at 9 13 04 PM

File: packages/line/tests/Line.test.js (hunk 1 above)
Comment (clcatalan) [reply packages/line/tests/Line.test.js:133]: unit test for hiding multiple series by default
Screenshot 2024-07-11 at 9 13 04 PM

File: packages/line/src/Line.js (hunk 2 above)
Comment (clcatalan) [reply packages/line/src/Line.js:110]: hi @plouc i've added to the documentation, however i'm just not sure what you mean by this needing to be added to the types, which file are you referring to? thank you!

File: website/src/data/components/line/props.ts [hunk 3]
```diff
@@ -541,6 +541,15 @@ const props: ChartProperty[] = [
...
         control: { type: 'switch' },
     },
+    {
+        key: 'initialHiddenIds',
+        flavors: allFlavors,
+        group: 'Interactivity',
+        help: `Hides certain series by default given their ids`,
+        type: 'boolean',
+        defaultValue: defaults.initialHiddenIds,
+        control: { type: 'switch' },
```
Comment (plouc) [inline website/src/data/components/line/props.ts:551]: No control should be defined, as the value cannot be changed via the demo.

File: website/src/data/components/line/props.ts [hunk 4]
```diff
@@ -541,6 +541,15 @@ const props: ChartProperty[] = [
         defaultValue: defaults.enableTouchCrosshair,
         control: { type: 'switch' },
     },
+    {
+        key: 'initialHiddenIds',
+        flavors: allFlavors,
+        group: 'Interactivity',
+        help: `Hides certain series by default given their ids`,
+        type: 'boolean',
```
Comment (plouc) [inline website/src/data/components/line/props.ts:549]: Invalid type.

**Review by plouc — CHANGES_REQUESTED** [review]

File: packages/line/src/Line.js (hunk 2 above)
Comment (plouc) [reply packages/line/src/Line.js:110]: Hi @ccatalan6, I'm referring to this file: https://github.com/plouc/nivo/blob/master/packages/line/index.d.ts

Comment (clcatalan) [reply packages/line/src/Line.js:110]: i've added to the types, let me know if there are concerns @plouc

File: website/src/data/components/line/props.ts (hunk 3 above)
Comment (clcatalan) [reply website/src/data/components/line/props.ts]: i've removed the control @plouc

File: packages/line/src/Line.js (hunk 2 above)
Comment (plouc) [reply packages/line/src/Line.js:110]: @clcatalan, thank you, the build is failing due to formatting issues, please run `make fmt`.

Comment (clcatalan) [reply packages/line/src/Line.js:110]: hi @plouc  i've run make fmt, but it seems the build is still failing? i checked out the ci, and im not quite sure how to resolve this
build failing

Comment (plouc) [reply packages/line/src/Line.js:110]: @clcatalan, this build was broken before your changes, it's unrelated, all good.

**Review by plouc — COMMENTED** [review]
LGTM

**Pull request merged**